Cli importer csv

cheburan opened the pull request:
Adding the converting tool to the CLI for the food-table CSV conversion to the package format.

lukashroch committed:
feat: personal access tokens

cheburan committed:
wip(cli): Adding CSV import command initial commit

cheburan committed:
wip(cli): parsing CSV file with Food records

cheburan committed:
wip(cli): converting CSV to the Inport Friendly Format

cheburan committed:
refactor(cli): Add convert-to-package as a separate  command

cheburan committed:
refactor(cli): Convertor for CSV - JSON files creation

cheburan committed:
fix(cli): Modifield CSV convertor to check for Food Categories disparety

cheburan committed:
refactor(cli): Adding debug outputs for GlobalFood csvToJson function

cheburan committed:
feat(workspace): Adde CLI terminal and minor changes in the convertor

cheburan committed:
feat(cli): added PSM import from CSV to Package Format

cheburan committed:
feat(cli): add additional parentCategories check and filtering before convertion to JSON, (via API request)

cheburan committed:
wip(cli): fix minoe bugs and added default psm for foods if not mentioned in CSV

cheburan committed:
docs(cli): adding basic docs for the convertor tool

cheburan commented:
[example.zip](https://github.com/user-attachments/files/15756917/example.zip)
Example of the files (CSV and JSON) to use with the cli convertor

cheburan committed:
chore(cli): rebased on main

cheburan committed:
refactor(cli): rebased to up-to-date main

cheburan committed:
fix(ui): se-cliboard bug fix - missing cliboardAvailable property

lukashroch commented on apps/api/src/http/controllers/admin/fdbs/category.controller.ts:
```diff
@@ -156,6 +157,14 @@ function adminCategoryController({
     res.json(categories);
   };
 
+  const all = async (
```
This might quickly become a perf hog and cause issues for larger sets, better use paginated query/response?

chrimyn234 replied:
Please correct me if I'm wrong, but it looks like this isn't being used anywhere. Also, it looks like browseMain already implements what this is intended to do but with pagination. Should it be removed?

lukashroch committed:
feat(fw): vue3 upgrade (V4-1249)

lukashroch committed:
Merge branch 'main' into cli_importer_CSV

cpra0021 committed:
feat(cli): packager also generates local-categories.json; some code refactoring and fixes

The script now also generates a local-categories.json. Fixed a few typos. Added additional logs.
Added trim() since to account for accidental whitespaces. Replaced some for...of loops with
Promise.all.

cpra0021 committed:
refactor(cli): replace for...of with promise.all; skip local foods not wanted by locale

Replace more for...of loops with promise.all so they can run in parallel. When creating local foods,
exclude foods that are not wanted by the locale.

cpra0021 committed:
style(cli): use arrowParens throughout for consistency

cpra0021 committed:
refactor(cli): update csv-import type of food composition record id

cpra0021 committed:
feat(cli): differentiate between associated food/category codes

Previously, the associated foods/categories were incomplete. Associated categories codes were
considered as food code even though they are not. Now, it correctly searches the database to
identify whether it is a food code or category code.

cpra0021 committed:
refactor(cli): optimize some functions

Use short circuiting; Replace array includes with set has for linear time complexity; Add more error
handling

cpra0021 committed:
perf(cli): refactor to use parallel/batch processing whenever applicable
